### PR TITLE
Create message to request messages and set message intervals

### PR DESCRIPTION
**Pull request by christian-aiello**
Wrote code to construct COMMAND_LONG (76) messages to send MAV_CMD_SET_MESSAGE_INTERVAL (511) and MAV_CMD_REQUEST_MESSAGE (512) commands.

### CODE REVIEW

File: flutter_app/lib/main.dart [hunk 1]
```diff
@@ -1,125 +1,81 @@
...
-  // This class is the configuration for the state. It holds the values (in this
-  // case the title) provided by the parent (in this case the App widget) and
-  // used by the build method of the State. Fields in a Widget subclass are
-  // always marked "final".
+  var sequence = 0;
+  var systemId = 255;
+  var componentId = 1;
 
-  final String title;
+  // example of requesting a single instance of a message using MAV_CMD_REQUEST_MESSAGE (512)
```
Comment (Xierumeng) [inline flutter_app/lib/main.dart]: Put this in a function to be called.

File: flutter_app/lib/main.dart [hunk 2]
```diff
@@ -1,125 +1,81 @@
...
+      confirmation: 0,
+      param1: desiredMessage.toDouble(),
+      param2: param2.toDouble(),
+      param3: 0,
+      param4: 0,
+      param5: 0,
+      param6: 0,
+      param7: 0);
+  var frm = MavlinkFrame.v1(sequence, systemId, componentId, commandLong);
+  parser.parse(frm.serialize());
```
Comment (Xierumeng) [inline flutter_app/lib/main.dart]: Is this working with Mission Planner?

Comment (christian-aiello) [reply flutter_app/lib/main.dart]: I haven't tested it with mission planner yet. Can mission planner receive messages over TCP?

File: flutter_app/lib/main.dart (hunk 1 above)
Comment (christian-aiello) [reply flutter_app/lib/main.dart]: Done.

File: flutter_app/lib/main.dart [hunk 3]
```diff
@@ -1,125 +1,91 @@
...
+        confirmation: 0,
+        param1: messageId.toDouble(),
+        param2: param2
+            .toDouble(), // parameter required for some messages, described in common message set
+        param3: 0,
+        param4: 0,
+        param5: 0,
+        param6: 0,
+        param7: 0);
+    var frm = MavlinkFrame.v1(sequence, systemId, componentId, commandLong);
```
Comment (Xierumeng) [inline flutter_app/lib/main.dart]: v2

File: flutter_app/lib/main.dart [hunk 4]
```diff
@@ -1,125 +1,91 @@
...
+  var componentId = 1;
+  var messageId = 33; // GLOBAL_POSITION_INT
+  var interval = 1000000;
 
-  // This class is the configuration for the state. It holds the values (in this
-  // case the title) provided by the parent (in this case the App widget) and
-  // used by the build method of the State. Fields in a Widget subclass are
-  // always marked "final".
+  // example of requesting a single instance of a message using MAV_CMD_REQUEST_MESSAGE (512)
+  Uint8List requestMessage(
```
Comment (Xierumeng) [inline flutter_app/lib/main.dart]: Return `MAVLinkFrame`

Comment (Xierumeng) [inline flutter_app/lib/main.dart]: Move both functions to another file.

File: flutter_app/lib/main.dart [hunk 5]
```diff
@@ -1,125 +1,91 @@
...
+        param5: 0,
+        param6: 0,
+        param7: 0);
+    var frm = MavlinkFrame.v1(sequence, systemId, componentId, commandLong);
+    return frm.serialize();
+  }
 
-  final String title;
+  // example of requesting messages at a specified rate using MAV_CMD_SET_MESSAGE_INTERVAL (511)
+  Uint8List setMessageInterval(int sequence, int systemId, int componentId,
```
Comment (Xierumeng) [inline flutter_app/lib/main.dart]: Same as above.

File: flutter_app/lib/main.dart [hunk 6]
```diff
@@ -1,125 +1,91 @@
...
+        param1: messageId.toDouble(),
+        param2: param2
+            .toDouble(), // parameter required for some messages, described in common message set
+        param3: 0,
+        param4: 0,
+        param5: 0,
+        param6: 0,
+        param7: 0);
+    var frm = MavlinkFrame.v1(sequence, systemId, componentId, commandLong);
+    return frm.serialize();
```
Comment (Xierumeng) [inline flutter_app/lib/main.dart]: Do not serialize, return frame.

File: flutter_app/lib/command_constructor.dart [hunk 7]
```diff
@@ -0,0 +1,73 @@
+import 'package:dart_mavlink/mavlink.dart';
+import 'package:dart_mavlink/dialects/common.dart';
+
+// ignore_for_file: avoid_print
+
+// Construct command to request a single instance of a message using MAV_CMD_REQUEST_MESSAGE (512)
+MavlinkFrame requestMessage(
+    int sequence, int systemId, int componentId, int messageId, int param2) {
```
Comment (Xierumeng) [inline flutter_app/lib/command_constructor.dart]: Add comment explaining what how each parameter is used (for callers of this function).

File: flutter_app/lib/command_constructor.dart [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+      param5: 0,
+      param6: 0,
+      param7: 0);
+  var frm = MavlinkFrame.v2(sequence, systemId, componentId, commandLong);
+  return frm;
+}
+
+// Construct command to request messages at a specified rate using MAV_CMD_SET_MESSAGE_INTERVAL (511)
+MavlinkFrame setMessageInterval(
+    int sequence, int systemId, int componentId, int messageId, int interval) {
```
Comment (Xierumeng) [inline flutter_app/lib/command_constructor.dart]: Same as above.

File: flutter_app/lib/command_constructor.dart [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+      }
+    }
+  });
+
+  parser.parse(requestMessage(sequence, systemId, componentId, messageId, 0)
+      .serialize());
+  parser.parse(
+      setMessageInterval(sequence, systemId, componentId, messageId, interval)
+          .serialize());
+}
```
Comment (Xierumeng) [inline flutter_app/lib/command_constructor.dart]: Move this to a new file: `flutter_app/test/[your test file name]`

File: flutter_app/lib/main.dart [hunk 10]
```diff
@@ -1,7 +1,9 @@
 import 'package:flutter/material.dart';
+import 'command_constructor.dart';
 
 void main() {
   runApp(const MyApp());
+  testConstruction();
```
Comment (Xierumeng) [inline flutter_app/lib/main.dart]: Should move this call in the test file as well, although I'm not sure specifically how to run it...

File: flutter_app/lib/command_constructor.dart (hunk 7 above)
Comment (christian-aiello) [reply flutter_app/lib/command_constructor.dart]: Done.

File: flutter_app/lib/command_constructor.dart (hunk 8 above)
Comment (christian-aiello) [reply flutter_app/lib/command_constructor.dart]: Done.

File: flutter_app/lib/command_constructor.dart (hunk 9 above)
Comment (christian-aiello) [reply flutter_app/lib/command_constructor.dart]: Done.

File: flutter_app/lib/main.dart (hunk 10 above)
Comment (christian-aiello) [reply flutter_app/lib/main.dart]: Setup testing with package:test and wrote tests. Can be run with `dart test` from flutter_app directory.

File: flutter_app/test/command_constructor_test.dart [hunk 11]
```diff
@@ -0,0 +1,46 @@
...
+import 'package:flutter_app/command_constructor.dart';
+import 'package:test/test.dart';
+
+void main() {
+  group('Command Constructor Tests', () {
+    var sequence = 0;
+    var systemId = 1;
+    var componentId = 0;
+    var messageId = 33; // GLOBAL_POSITION_INT
+    var interval = 1000000;
```
Comment (Xierumeng) [inline flutter_app/test/command_constructor_test.dart]: Add a comment that this is in microseconds and represents 1 second (e.g. `// 1000000 microseconds == 1 second` ).

File: flutter_app/test/command_constructor_test.dart [hunk 12]
```diff
@@ -0,0 +1,46 @@
...
+
+    var dialect = MavlinkDialectCommon();
+
+    test('Request Message', () {
+      var parser = MavlinkParser(dialect);
+      parser.stream.listen((MavlinkFrame frm) {
+        if (frm.message is CommandLong) {
+          var cl = frm.message as CommandLong;
+          expect(cl.command, equals(512));
+          expect(cl.param1, equals(33));
```
Comment (Xierumeng) [inline flutter_app/test/command_constructor_test.dart]: Pull magic numbers into variables.
```dart
// Setup
expected_command = 512;
expected_id = 33;
...
expect(c1.command, equals(expected_command));
expect(c1.param1, equals(expected_id));
```

File: flutter_app/test/command_constructor_test.dart [hunk 13]
```diff
@@ -0,0 +1,46 @@
...
+    });
+
+    test('Set Message Interval', () {
+      var parser = MavlinkParser(dialect);
+      parser.stream.listen((MavlinkFrame frm) {
+        if (frm.message is CommandLong) {
+          var cl = frm.message as CommandLong;
+          expect(cl.command, equals(511));
+          expect(cl.param2, equals(1000000));
+          expect(cl.param1, equals(33));
```
Comment (Xierumeng) [inline flutter_app/test/command_constructor_test.dart]: Same here with magic numbers.

File: flutter_app/test/command_constructor_test.dart [hunk 14]
```diff
@@ -0,0 +1,47 @@
...
+import 'package:dart_mavlink/dialects/common.dart';
+import 'package:flutter_app/command_constructor.dart';
+import 'package:test/test.dart';
+
+void main() {
+  group('Command Constructor Tests', () {
+    var sequence = 0;
+    var systemID = 1;
+    var componentID = 0;
+    var messageID = 33; // GLOBAL_POSITION_INT
```
Comment (Xierumeng) [inline flutter_app/test/command_constructor_test.dart:11]: Make these `const` .

Comment (Xierumeng) [inline flutter_app/test/command_constructor_test.dart]: Make these `const` .

File: flutter_app/test/command_constructor_test.dart [hunk 15]
```diff
@@ -0,0 +1,47 @@
...
+    var interval = 1000000; // Time interval in microseconds == 1 second
+    var requestMessageCommandNumber = 512; // MAV_CMD (MavLink Command) Number for requesting a single instance of a particular MAVLink message ID
+    var setIntervalCommandNumber = 511; // MAV_CMD (MavLink Command) Number for setting an interval between messages for a particular MAVLink message ID
```
Comment (Xierumeng) [inline flutter_app/test/command_constructor_test.dart]: Move these into the appropriate test.

File: flutter_app/test/command_constructor_test.dart [hunk 16]
```diff
@@ -0,0 +1,47 @@
...
+    var dialect = MavlinkDialectCommon();
```
Comment (Xierumeng) [inline flutter_app/test/command_constructor_test.dart]: Make this `const` and move it to the top.

**Review by Xierumeng — COMMENTED** [review]
Reviewed.

File: flutter_app/test/command_constructor_test.dart (hunk 14 above)
Comment (christian-aiello) [reply flutter_app/test/command_constructor_test.dart]: Done.

File: flutter_app/test/command_constructor_test.dart (hunk 15 above)
Comment (christian-aiello) [reply flutter_app/test/command_constructor_test.dart]: Done.

File: flutter_app/test/command_constructor_test.dart (hunk 16 above)
Comment (christian-aiello) [reply flutter_app/test/command_constructor_test.dart]: I don' think I can. "The constructor being called isn't a const constructor."

File: flutter_app/test/command_constructor_test.dart [hunk 17]
```diff
@@ -0,0 +1,49 @@
...
+import 'package:flutter_app/command_constructor.dart';
+import 'package:test/test.dart';
+
+void main() {
+  group('Command Constructor Tests', () {
+    const sequence = 0;
+    const systemID = 1;
+    const componentID = 0;
+    const messageID = 33; // GLOBAL_POSITION_INT
+    var dialect = MavlinkDialectCommon();
```
Comment (Xierumeng) [inline flutter_app/test/command_constructor_test.dart]: Move this to the top?

File: flutter_app/test/widget_test.dart
Comment (Xierumeng) [inline flutter_app/test/widget_test.dart:1]: Undelete this, it will be cleaned up in a different task.

**Review by Xierumeng — COMMENTED** [review]
Reviewed.

**Review by Xierumeng — APPROVED** [review]
Approved.

**Pull request merged**